test/unit/pages/patientdata.test.js

```diff
@@ -132,87 +138,81 @@ describe('PatientData', function () {
-    var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-    expect(elem).to.be.ok;
-    
-    var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
-    expect(x).to.be.ok;
-  });
+      // Try out using the spread props syntax in JSX
+      var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
+      expect(elem).to.be.ok;
+      
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+    });
 
-  it ('should render when data is present for current patient', function() {
-    var props = {
-      timePrefs: {
-        timezoneAware: false,
-        timezoneName: null
-      },
-      patient: {
-        userid: 40
-      },
-      patientData: {
-        40: { data: [ 1, 2] }
-      },
-      fetchingPatient: false,
-      fetchingPatientData: false,
-      queryParams: {},
-      trackMetric: sinon.stub()
-    };
-
-    PatientData.__set__('Modal', React.createClass({
-      render: function() {
-        console.log('Mock', this.props);
-        return <div className='fake-modal-view'/>;
-      }
-    }));
-
-    // Try out using the spread props syntax in JSX
-    var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-    expect(elem).to.be.ok;
-    
-    // var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'fake-modal-view');
-    // expect(x).to.be.ok;
+    it ('should render the no data message when no data is present for current patient', function() {
+      var props = {
+        timePrefs: {
+          timezoneAware: false,
+          timezoneName: null
+        },
+        patient: {
+          userid: 40
+        },
+        patientData: {
+          41: { data: [] }
+        },
+        fetchingPatient: false,
+        fetchingPatientData: false,
+        queryParams: {},
+        trackMetric: sinon.stub()
+      };
+
+      // Try out using the spread props syntax in JSX
+      var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
+      expect(elem).to.be.ok;
+      
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+    });
+
+    it ('should render when data is present for current patient', function() {
+      var props = {
+        timePrefs: {
+          timezoneAware: false,
+          timezoneName: null
+        },
+        patient: {
+          userid: 40
+        },
+        patientData: {
+          40: { data: [ 1, 2] }
+        },
+        fetchingPatient: false,
+        fetchingPatientData: false,
+        queryParams: {},
+        trackMetric: sinon.stub()
+      };
+
+      var pdElem = React.createElement(PatientData, props);
+      var elem = TestUtils.renderIntoDocument(pdElem);
+      expect(elem).to.be.ok;
+      
+      // var x = TestUtils.scryRenderedDOMComponentsWithClass(elem, 'fake-modal-view');
+      // console.log(x.length);
+      //expect(x).to.be.ok;
```

Comment:
Glad you spotted this. One of things I was trying out whilst testing patientdata was mocking other modules that are dependencies in patientdata.js, specifically chart/modal.js. 

In order to do this I looked into a number of libraries that essentially allows you to monkey path CommonJS modules and the functionality of require. For some reason though after using rewire.js for this purpose, when rendering PatientData it somehow bypasses the rewiring and uses the original dependency - therefore resulting the commented out code failing. I was wondering:

 a) whether you have had any experience mocking dependencies when working with Webpack.js
 b) whether you could shed any light on why the desired behaviour is not working.

[Here is a blog post that discusses the approach](http://substantial.com/blog/2014/11/11/test-driven-react-how-to-manually-mock-components/)